tests/Mqtt5ClientCredentialsTest.cpp

```diff
@@ -0,0 +1,1185 @@
+AWS_STATIC_STRING_FROM_LITERAL(s_mqtt5_test_envName_iot_pkcs11_pin, "AWS_TEST_PKCS11_PIN");
+AWS_STATIC_STRING_FROM_LITERAL(s_mqtt5_test_envName_iot_pkcs11_private_key_label, "AWS_TEST_PKCS11_PKEY_LABEL");
+AWS_STATIC_STRING_FROM_LITERAL(s_mqtt5_test_envName_iot_pkcs11_cert, "AWS_TEST_PKCS11_CERT_FILE");
+AWS_STATIC_STRING_FROM_LITERAL(s_mqtt5_test_envName_iot_pkcs11_ca, "AWS_TEST_PKCS11_CA_FILE");
+// C++ specific PKCS11 check: only runs PKCS11 if 'DUSE_OPENSSL=ON' is set in the builder
+AWS_STATIC_STRING_FROM_LITERAL(s_test_envName_iot_pkcs11_use_openssl, "AWS_TEST_PKCS11_USE_OPENSSL_SET");
+
+AWS_STATIC_STRING_FROM_LITERAL(s_mqtt5_test_envName_iot_pkcs12_key, "AWS_TEST_MQTT5_IOT_CORE_PKCS12_KEY");
+AWS_STATIC_STRING_FROM_LITERAL(
+    s_mqtt5_test_envName_iot_pkcs12_key_password,
+    "AWS_TEST_MQTT5_IOT_CORE_PKCS12_KEY_PASSWORD");
+
+AWS_STATIC_STRING_FROM_LITERAL(s_mqtt5_test_envName_iot_windows_cert, "AWS_TEST_MQTT5_IOT_CORE_WINDOWS_CERT_STORE");
+
+AWS_STATIC_STRING_FROM_LITERAL(
+    s_mqtt5_test_envName_iot_credential_access_key,
+    "AWS_TEST_MQTT5_ROLE_CREDENTIAL_ACCESS_KEY");
+AWS_STATIC_STRING_FROM_LITERAL(
+    s_mqtt5_test_envName_iot_credential_secret_access_key,
+    "AWS_TEST_MQTT5_ROLE_CREDENTIAL_SECRET_ACCESS_KEY");
+AWS_STATIC_STRING_FROM_LITERAL(
+    s_mqtt5_test_envName_iot_credential_session_token,
+    "AWS_TEST_MQTT5_ROLE_CREDENTIAL_SESSION_TOKEN");
+
+AWS_STATIC_STRING_FROM_LITERAL(s_mqtt_cred_access_key, "AWS_ACCESS_KEY_ID");
+AWS_STATIC_STRING_FROM_LITERAL(s_mqtt_cred_secret_access_key, "AWS_SECRET_ACCESS_KEY");
+AWS_STATIC_STRING_FROM_LITERAL(s_mqtt_cred_session_token, "AWS_SESSION_TOKEN");
+
+AWS_STATIC_STRING_FROM_LITERAL(s_mqtt5_test_envName_iot_cognito_endpoint, "AWS_TEST_MQTT5_COGNITO_ENDPOINT");
+AWS_STATIC_STRING_FROM_LITERAL(s_mqtt5_test_envName_iot_cognito_identity, "AWS_TEST_MQTT5_COGNITO_IDENTITY");
+
+AWS_STATIC_STRING_FROM_LITERAL(s_mqtt5_test_envName_iot_profile_credentials, "AWS_TEST_MQTT5_IOT_PROFILE_CREDENTIALS");
+AWS_STATIC_STRING_FROM_LITERAL(s_mqtt5_test_envName_iot_profile_config, "AWS_TEST_MQTT5_IOT_PROFILE_CONFIG");
+
+AWS_STATIC_STRING_FROM_LITERAL(s_mqtt5_test_envName_iot_x509_endpoint, "AWS_TEST_MQTT5_IOT_CORE_X509_ENDPOINT");
+AWS_STATIC_STRING_FROM_LITERAL(s_mqtt5_test_envName_iot_x509_role_alias, "AWS_TEST_MQTT5_IOT_CORE_X509_ROLE_ALIAS");
+AWS_STATIC_STRING_FROM_LITERAL(s_mqtt5_test_envName_iot_x509_thing_name, "AWS_TEST_MQTT5_IOT_CORE_X509_THING_NAME");
+AWS_STATIC_STRING_FROM_LITERAL(s_mqtt5_test_envName_iot_x509_cert, "AWS_TEST_MQTT5_IOT_CORE_X509_CERT");
+AWS_STATIC_STRING_FROM_LITERAL(s_mqtt5_test_envName_iot_x509_key, "AWS_TEST_MQTT5_IOT_CORE_X509_KEY");
+AWS_STATIC_STRING_FROM_LITERAL(s_mqtt5_test_envName_iot_x509_ca, "AWS_TEST_MQTT5_IOT_CORE_X509_CA");
+
+// Needed to return "success" instead of skip in Codebuild so it doesn't count as a failure
+AWS_STATIC_STRING_FROM_LITERAL(s_mqtt5_test_envName_codebuild, "CODEBUILD_BUILD_ID");
```

Comment:
Just realized that the `SKIPPED` tests will be treated as failure for ctest, sorry that it caused the issue here. 
The issue is fixed in CMake 3.9, while we are using 3.1 as the minimum requirement cross the SDKs. This is just a note that we could probably remove the check when we bump the minimum support for CMake to 3.9.